docs/content/ingestion/tasks.md

```diff
@@ -89,9 +89,17 @@ The Index Task is a simpler variation of the Index Hadoop task that is designed
 |property|description|required?|
 |--------|-----------|---------|
 |type|The task type, this should always be "index".|yes|
-|id|The task ID. If this is not explicitly specified, Druid generates the task ID using the name of the task file and date-time stamp. |no|
+|id|The task ID. If this is not explicitly specified, Druid generates the task ID using task type, data source name, interval, date-time stamp and priority of the task. |no|
 |spec|The ingestion spec. See below for more details. |yes|
 
+#### Task Priority
+
+Task priority is used for acquiring a lock on an interval for a datasource. Tasks with higher priority can preempt lower-priority tasks for the same datasource and interval if ran concurrently.
+
+Priority order for different task types - Realtime Index Task > Hadoop/Index Task > Merge/Append Task > Other Tasks. Tasks of same type cannot preempt one another.
```

Comment:
1) do we need to mention that this is default priority and can be overrided by using task context ?  
2) Minor nit : rename to "tasks of same priority cannot preempt one another"